### PR TITLE
Move to YoastCS 2.0 and bring in line with other plugins autoloading

### LINKED ISSUE #62: Undefined index in WordPress_Plugin_Version_Control.php
The issue is found during testing of 13.2 release candidates. For Site Health feature upgrade routines should be tested.

### How can we reproduce this behavior?
1. Download zip, install and activate Yoast Test Helper 1.4.
2. Install and activate Yoast SEO 13.2-RC1 
3. Go to Yoast Test Helper

`Notice: Undefined index: version in /srv/www/local/public_html/wp-content/plugins/yoast-test-helper/src/WordPress_Plugin_Version_Control.php on line 159
`

Screenshot 2020-02-20 at 11 25 34

### LINKED ISSUE #63: Save the permalinks structure after changing custom post types
## Describe the feature you'd like

When the setting `Enable post types & taxonomies.` is changed, also update the permalink structure in WordPress, preventing 404s in the docker environment.

## Reproduce

1. Disable `Enable post types & taxonomies.`.
1. Save the WP permalink settings.
1. Enable `Enable post types & taxonomies.`.
1. Go to a frontend page of a book for example and see a 404 page. Tip, use the sitemap: http://basic.wordpress.test/book-sitemap.xml

The workaround is manually saving the permalink settings again.

### PR DESCRIPTION
**Pull request by jdevalk**
## Summary

This PR can be summarized in the following changelog entry:

* Moves the entire plugin to YoastCS 2.0.
* Implemented `composer check-cs` and `composer fix-cs` scripts.
* Fixes a notice on the admin page when a plugin wasn't installed.
* Flushes the rewrite rules after enabling the books and movies post type.

## Relevant technical choices:

* Moved away from PSR_4 autoloading to classmap, so we could be in line with other plugins.
* Changed the namespace from `Yoast\Test_Helper` to `Yoast\WP\Test_Helper` to be in line with other plugins.

## Milestone

* [x] I've attached the next release's milestone to this pull request.

## Test instructions

This PR can be tested by following these steps:

* Well, everything should still work as expected, basically.
* Also, test #63.

Fixes #62
Fixes #63

### CODE REVIEW

**Comment by jdevalk** [comment]
@jrfnl @moorscode for some reason in the build the files still have the old names (PSR-4 style), even though in the branch they're renamed... Github fluke?

File: src/admin-debug-info.php [hunk 1]
```diff
@@ -79,6 +75,6 @@ public function handle_submit() {
 			$this->option->set( 'show_options_debug', isset( $_POST['show_options_debug'] ) );
 		}
 
-		wp_safe_redirect( self_admin_url( 'tools.php?page=' . apply_filters( 'yoast_version_control_admin_page', '' ) ) );
+		wp_safe_redirect( self_admin_url( 'tools.php?page=' . apply_filters( 'Yoast\WP\Test_Helper\admin_page', '' ) ) );
```
Comment (jrfnl) [inline src/admin-debug-info.php]: Is this test helper available publicly ? If so, the old filter name should be deprecated.
See: https://github.com/Yoast/Wiki/wiki/How-to-rename-a-hook

:point_up_2: This question applies to all renamed hooks in the test helper.

File: src/admin-notifications.php [hunk 2]
```diff
@@ -56,7 +52,7 @@ public function display_notifications() {
 		echo '<div style="margin: 15px 0 15px -15px;">';
 		foreach ( $notifications as $notification ) {
 			// phpcs:ignore WordPress.XSS.EscapeOutput.OutputNotEscaped
-			echo '<div class="notice notice-' . esc_attr( $notification->get_type() ) . '"><p>' . $notification->get_message() . '</p></div>';
+			echo '<div class="notice notice-' . esc_attr( $notification->get_type() ) . '"><p>' . esc_html( $notification->get_message() ) . '</p></div>';
```
Comment (jrfnl) [inline src/admin-notifications.php]: Are these messages always without HTML ? Or should `wp_kses_post()` be used here ?

File: src/inline-script.php [hunk 3]
```diff
@@ -81,11 +78,11 @@ public function get_controls() {
 	public function handle_submit() {
 		if ( check_admin_referer( 'yoast_seo_test_inline_script' ) !== false ) {
 			$this->option->set( 'add_inline_script', isset( $_POST['add_inline_script'] ) );
-			$this->option->set( 'inline_script_handle', (string) $_POST['inline_script_handle'] );
-			$this->option->set( 'inline_script', stripslashes( (string) $_POST['inline_script'] ) );
+			$this->option->set( 'inline_script_handle', filter_input( INPUT_POST, 'inline_script_handle', FILTER_SANITIZE_STRING ) );
+			$this->option->set( 'inline_script', filter_input( INPUT_POST, 'inline_script', FILTER_SANITIZE_STRING ) );
```
Comment (jrfnl) [inline src/inline-script.php]: Using `FILTER_SANITIZE_STRING` is next to useless.

File: src/post-types.php [hunk 4]
```diff
@@ -153,22 +152,37 @@ public function handle_submit() {
...
+
 	/**
 	 * Sets a boolean option based on a POST parameter.
 	 *
 	 * @param string $option The option to check and set.
 	 */
 	private function set_bool_option( $option ) {
 		// The nonce is checked in the handle_submit function.
-		// phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
+		// @phpcs:ignore WordPress.Security.NonceVerification.Missing
```
Comment (jrfnl) [inline src/post-types.php]: The `@` is not needed.
```suggestion
		// phpcs:ignore WordPress.Security.NonceVerification.Missing
```

File: src/schema.php [hunk 5]
```diff
@@ -103,10 +99,10 @@ public function handle_submit() {
 			$this->option->set( 'replace_schema_domain', isset( $_POST['replace_schema_domain'] ) );
 		}
 
-		$this->option->set( 'is_needed_breadcrumb', $_POST['is_needed_breadcrumb'] );
-		$this->option->set( 'is_needed_webpage', $_POST['is_needed_webpage'] );
+		$this->option->set( 'is_needed_breadcrumb', isset( $_POST['is_needed_breadcrumb'] ) ? true : false );
```
Comment (jrfnl) [inline src/schema.php]:
```suggestion
		$this->option->set( 'is_needed_breadcrumb', isset( $_POST['is_needed_breadcrumb'] ) );
```

File: src/schema.php [hunk 6]
```diff
@@ -103,10 +99,10 @@ public function handle_submit() {
...
+		$this->option->set( 'is_needed_webpage', isset( $_POST['is_needed_webpage'] ) ? true : false );
```
Comment (jrfnl) [inline src/schema.php]:
```suggestion
		$this->option->set( 'is_needed_webpage', isset( $_POST['is_needed_webpage'] ) );
```

File: phpcs.xml.dist [hunk 7]
```diff
@@ -4,22 +4,35 @@
...
-        <exclude name="PHPCompatibility.PHP.NewKeywords.t_namespaceFound" />
-        <exclude name="PHPCompatibility.PHP.NewKeywords.t_useFound" />
-        <exclude name="PHPCompatibility.PHP.NewKeywords.t_dirFound" />
-        <exclude name="PHPCompatibility.PHP.NewClosure.Found" />
-        <exclude name="Yoast.Files.FileName" />
-        <exclude name="WordPress.VIP" />
+        <properties>
+   			<!-- Provide the plugin specific prefixes for all naming related sniffs. -->
+   			<property name="prefixes" type="array">
+   				<element value="Yoast\WP\Test_Helper"/>
```
[the remark this answers is not in the document]
Comment (jdevalk) [reply phpcs.xml.dist:17]: Nope.

File: src/admin-debug-info.php (hunk 1 above)
Comment (jdevalk) [reply src/admin-debug-info.php:78]: It wasn't before this.

File: composer.json [hunk 8]
```diff
@@ -10,18 +10,31 @@
...
+    ]
   },
   "config": {
     "platform": {
       "php": "5.6"
     }
   },
   "require-dev": {
-    "yoast/yoastcs": "^0.5",
-    "roave/security-advisories": "dev-master",
```
[the remark this answers is not in the document]
Comment (jdevalk) [reply composer.json:24]: Brought it back.

**Comment by jdevalk** [comment]
Thx for all your input on this @jrfnl. Going to merge this, I think the only thing left is:

* The .gitignore file is missing ignores for the typical PHPCS overload files.
* As a side-note: the ruleset could do with minor syncing with the other plugin rulesets as for the inline documentation, but if you like I'll fix that up next week.

If you could do those in a separate PR later that'd be appreciated!

**Comment by jrfnl** [comment]
@jdevalk :+1: I'll add this plugin to my regular scan round now and will send in additional PRs where needed.

**Pull request merged**